packages/client/carl.lua

```diff
@@ -1,373 +1,106 @@
---- @field private cache table?
-local Repositories = { cache = nil }
-
---- Updates the repositories library with the provided repository.
---- @param name string
---- @param url string
-function Repositories:set(name, url)
-    self.cache[name] = url
-    self:save()
-end
-
---- Get a repository url by name
---- @param name string
---- @return string?
-function Repositories:get(name)
-    return self.cache[name]
-end
-
---- Load the repositories from disk.
-function Repositories:load()
-    local file = fs.open(REPOSITORIES_FILE, "r")
-    local line = file.readLine()
-    self.cache = {}
-
-    while line ~= nil do
-        local name, url = line:match("([^ ]+) ([^ ]+)")
-        self.cache[name] = url
-
-        line = file.readLine()
-    end
-
-    file.close()
-end
-
---- Get the repository map.
---- @return table
-function Repositories:all()
-    return self.cache
-end
-
---- Save the repositories file to disk.
-function Repositories:save()
-    local file = fs.open(REPOSITORIES_FILE, "w")
-
-    for name, url in pairs(self.cache) do
-        file.write(name .. " " .. url)
+    term.setTextColour(colours.white)
+    if _G.type(message) == "string" then
+        ---@diagnostic disable-next-line: param-type-mismatch
+        message = message:format(table.unpack(args or {}))
     end
 
-    file.close()
-end
-
---- Remove the repository from the library.
---- @param name string
-function Repositories:remove(name)
-    --- @diagnostic disable-next-line: param-type-mismatch
-    self:set(name, nil)
-end
-
--- * Command Handling
-
---- Bootstrap function to be run on shell startup
-local function bootstrap()
-    for _, entry in ipairs(Manifest:all()) do
-        tryAddAlias(entry)
-    end
+    print(" " .. message)
 end
 
-Manifest:load()
-Repositories:load()
+-- * Command handling
 
 local command = ...
 
-if command == "install" then
-    -- todo array
-    -- todo protect against conflicts
-
-    local pkg = arg[2]
-    local repo = pkg:sub(1, pkg:find("/") - 1)
-
-    message("info", "PKG", ("Resolving \"%s\""):format(pkg))
-
-    local pkg_data = apiRequest("/pkg/" .. pkg, {
-        definitionURL = Repositories:get(repo)
-    })
-
-    if pkg_data == nil then
-        return
-    end
-
-    message("success", "PKG",
-        ("Found v%s with %d file%s"):format(pkg_data["version"], #pkg_data["files"],
-            #pkg_data["files"] == 1 and "" or "s"))
-
-    local entry = ManifestEntry:new(pkg_data["name"], pkg_data["version"], pkg_data["cli"], pkg_data["repo"])
-
-    local pkg_dir = entry:getDir()
-    fs.makeDir(pkg_dir)
-
-    for _, file in ipairs(pkg_data["files"]) do
-        local path = pkg_dir .. "/" .. file["path"]
-
-        message("info", "DWN", file["path"])
-        downloadFile(file["url"], path)
-        message("success", "DWN", ("Downloaded %s (%dB)"):format(file["path"], fs.getSize(path)))
+local function handle()
+    local carl = require("init")
+
+    carl.setLogHandler(function(record)
```

Comment:
Do you think we should default this to printing to console, and then allow it to be overwritten?

Reply:
I don't mind! I personally think it makes more sense for it to be unset initially to avoid unexpected printing when developers require the module. That way it will just behave like a normal function but developers can still hook into the logs if they want to!

Reply:
I've talked to @Mcrevs who agrees that for simple programs it would be much more convenient for there to be a default log handler. I will move the current handler into `init.lua` to achieve this!